raiden/network/rpc/client.py

```diff
@@ -474,7 +495,14 @@ def approve(self, contract_address, allowance):
             startgas=self.startgas,
             gasprice=self.gasprice,
         )
-        self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+
+        try:
+            self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+        except JSONRPCPollTimeoutException:
+            raise BlockchainPollTimeout()
+        except InvalidTransaction:
+            raise BlockchainInvalidTransaction()
```

Comment:
wouldn't it be nicer for debugging purposes to just "re-raise"? 
```
except InvalidTransaction as e:
  raise e
```
Then there would be at least a chance to learn about the kind of invalid transaction.